### DISCUSSION TITLE
How To Display Saved Data To Kanban View Based On Related Field.

### DISCUSSION THREAD

**Discussion by it12uw** [OCA/odoo-community.org#112] [General]
I have 3 different kanban views,
The first is the Kanban view to display block data using the block.py model (blok.mesin) where there are 5 blocks of data from blocks, namely Blocks E, F, G, H, and I. The second is the Kanban view to display machine lines using the line model. py (line.mesin.produksi) which has 13 lines. and the last one is the Kanban view for production machines using the datamesin.py model (data.mesin.produksi).

I input it into the machine data form based on the fields below.

    divisi_id = fields.Many2one('hr.department', string="Divisi")
    block = fields.Many2one('block.mesin', string='Block')
    line= fields.Many2one('line.mesin.produksi', string='Line')
    nomor_mesin = fields.Char("Nomor Mesin",tracking=True)
    mesin = fields.Char(string="Nama Mesin",tracking=True)
    kode_kain = fields.Selection([
        ('C415','C415'),
        ('C405','C405'),
        ('C394','C394'),
        ('C222','C222'),
        ('C437','C437'),
        ('UMC403','UMC403'),
        ('PE408','PE408'),
        ('PE411','PE411'),
        ('N1125-38.5','N1125-38.5'),
        ('N1125-30','N1125-30'),
        ('N1125','N1125'),
        ('C2 642-SB UR','C2 642-SB UR'),
        ('C3752-SB UR','C3752-SB UR'),
        ('N3936','N3936'),
        ('R396','R396'),
        ('UMR 08','UMR 08'),
        ('UMR 07','UMR 07'),
        ('TR 3746-RJ','TR 3746-RJ'),
        ('TR 3746DESTEX','TR 3746DESTEX'),
        ('UMR 04','UMR 04'),
        ('UMR 06','UMR 06'),
        ('C401-SB UR','C401-SB UR'),
    ], string="Fabric Code")

   rpm = fields.Integer(string='RPM')

For example, the data I created is as follows:

Nomor Mesin: I25
Nama Mesin: Shuttle
Kode Kain: UMR 07
RPM: 5000
Line: Line 1
Block: Block I
Divisi: Sales

My goal is that when I save the record, the data will appear in Blok I and Line 1.
So when the user accesses the Kanban view and clicks on Block I and Line 1, the data appears there.
![dashboard for Blok ](https://github.com/OCA/odoo-community.org/assets/135787412/6d8891f6-2bff-475e-8fb6-b17f3251ea83)
![dashboard for Line](https://github.com/OCA/odoo-community.org/assets/135787412/983f6613-38ae-4472-bb14-c7309f0f2096)

**the problem is everytime i try save the record, it appear in every Blok and Line**

any idea how to achieve my goal?

**Comment by Shide** [comment]
On the action of the record, put a domain to filter what records do you want to see.
For example:
```
<record id="action_view_account_move_update_analytic" model="ir.actions.act_window">
        <field name="name">Update Analytic for selected Account Moves</field>
        <field name="res_model">account.move.update.analytic.wizard</field>
        <field name="view_mode">form</field>
        <field name="domain">[('line_id', '=', active_id)]</field>
        <field name="context">{'default_line_id': active_id}</field>
        <field name="target">new</field>
    </record>
```

If you can't launch an action, you can add extra parameters to a context in the form or kanban view:
The first one (search_default_xxxx) to filter the record on the search bar, the second one (default_xxx) to put a default value of the parent if any record is created from the target view.
```
context="{'search_default_partner_id': active_id, 'default_partner_id': active_id}"
```

Also you can add to a context the view you want to load when clicking the element:
```
<field name="product_stock_quant_ids" context="{'kanban_view_ref': 'stock_barcode.stock_quant_barcode_kanban_2', 'tree_view_ref': 'stock_barcode.view_stock_quant_tree'}"/>
```

_Examples above are copied directly get from Odoo codebase._

**Reply by it12uw** [reply]
i found this error
Error:


Traceback:
Error: NameError: name 'active_id' is not defined
    at PY_ensurepy (http://localhost:8069/web/content/2226-febd5b3/web.assets_backend.js:88:65)
    at py.evaluate (http://localhost:8069/web/content/2226-febd5b3/web.assets_backend.js:205:8)
    at py.evaluate (http://localhost:8069/web/content/2226-febd5b3/web.assets_backend.js:212:111)
    at py.evaluate (http://localhost:8069/web/content/2226-febd5b3/web.assets_backend.js:214:99)
    at py.eval (http://localhost:8069/web/content/2226-febd5b3/web.assets_backend.js:219:284)
    at http://localhost:8069/web/content/2226-febd5b3/web.assets_backend.js:365:136
    at _.each._.forEach (http://localhost:8069/web/content/2005-66856fa/web.assets_common.js:108:558)
    at _.<computed> [as each] (http://localhost:8069/web/content/2005-66856fa/web.assets_common.js:165:526)
    at eval_domains (http://localhost:8069/web/content/2226-febd5b3/web.assets_backend.js:364:265)
    at Object.pyeval [as eval] (http://localhost:8069/web/content/2226-febd5b3/web.assets_backend.js:389:24)
